### ISSUE TITLE
0.19 release checklist

### ISSUE THREAD

**Issue by lynxlynxlynx** [ja2-stracciatella/ja2-stracciatella#1356]
Lest I forget, we have a cleaner version of the release checklist in docs/.

## Pre-release development
- [x] Fix or postpone issues in the [milestone](https://github.com/ja2-stracciatella/ja2-stracciatella/milestone/7)
- [x] Triage open pull requests and recent bugs
- [x] Finalise choice of version number
  - [x] Decide if this version should be 1.0 #443 
- [x] Add new contributors to contributors.txt (make rebuild-contributors-list)
- [x] Update changes.md (run docs/draft-changelog.sh to get started)
- [x] Update man page if necessary

## Release candidate
- [x] Optionally update version number to `-rc` and tag
- [x] Mark the tag as a Github prerelease
- [x] Runtime tests of prerelease packages
  - [x] Linux
  - [x] Windows
  - [x] OS X
  
Runtime tests:
 * IMP creation
 * AIM and extending contracts
 * taking Drassen airport or more
 * buying from Bobby Ray's
 * visual inspection of laptop screens

If you want to build the packages manually:

    $ make clean
    $ make build-releases # to build all releases
    $ make build-win-release-on-linux # crosscompile Windows release on Linux
    $ make build-release-on-mac # build the mac bundle
    $ make package-appimage # build the appimage

# Release
- [x] Attach built versions to GitHub release
  - [x] Linux, Windows
  - [x] OS X
- [x] Document new features on the website
- [x] Make GitHub prerelase the release, update texts
- [x] Announce
  - [x] Write main announcement
    - mention c0281336fda
  - [x] Website download page (update frontmatter variables as needed)
  - [x] Website news
  - [x] Bear's pit
  - [x] Moddb

## Post-release
- [x] Set version to +1 and add the `-dev` suffix
- [x] Close this bug and GitHub milestone
- [x] Create new milestone if none exists yet

**Comment by selaux** [comment]
As it was mentioned in Discord: Should we prioritize the issues currently in the `0.19` milestone and focus on the next release?

**Comment by lynxlynxlynx** [comment]
Yep. I got distracted while waiting for feedback.

**Comment by lynxlynxlynx** [comment]
Time flies! I thought it's been 5 years since the project has been revived in the current form (first commit here on 5.2.2016) and that it'd be nice to mention it in the release notes, but that's actually 6 already. 😃

**Comment by lynxlynxlynx** [comment]
I wrote a draft of the announcement here:
https://github.com/ja2-stracciatella/ja2-stracciatella.github.io/blob/master/_posts/_future-release-0.19.0.md

Can lift the changes.md update from there, but wanted to wait for us to be done first.

Also, I think the long changelog would look better with subsections per type. Compare https://ja2-stracciatella.github.io/2021/04/23/release-0.18.0.html to not having the type (prefix) in each line, but the long list split into several. What do you think? We can keep it the same old style in changes.md.

**Comment by ja2-derek** [comment]
I think there are some `daattali` issue numbers on the 0.19.0.md by mistake. I don't have strong opinion on sub-section vs prefix. The repeated prefix does not bother me. The problem is the length of the list - which means we should have more frequent releases!

**Comment by lynxlynxlynx** [comment]
The daattali thing is just a github display artifact, perhaps from the original jekyll theme. It should render as normal numbers on the actual site.

**Comment by selaux** [comment]
I created the new release tag. Lets see if builds work.

**Comment by lynxlynxlynx** [comment]
Didn't try the builds, but testing from git directly didn't show any problems. Played until rebuking the counterattack on Dressen.

**Comment by lynxlynxlynx** [comment]
The appimage mostly works fine. Using the same save I was testing with earlier, I did get the soft "hour" lockup twice during combat, but that was it.

**Comment by ja2-derek** [comment]
What were you doing when you got the soft lockup?

**Comment by lynxlynxlynx** [comment]
I was shooting, it felt like potentially a hidden interrupt. Once it happened while killing the last enemy in a Meduna sector, so I then had to end the turn for things to autoresolve. Nothing tangible really, but I did experience it later with a dev branch as well. Weird stuff.

**Comment by lynxlynxlynx** [comment]
Ok, found one case, but I still very much doubt that it is new. If you mortar a soldier without killing him, they tend to move as usual. Targetting them with a gun doesn't show any body position, just the CTH (I have that enabled). You can aim normally still. Unfortunately loading a save mad just after getting the bad cursor restores normal state and everything works fine.

**Comment by selaux** [comment]
I've tested now the Appimage on Linux, Android App and macOS App.

I found one small mistake with the German Translation for the SaveLoadScreen which I think makes sense to fix in the release.

macOS is a different beast.

- Scaling issues: When starting in windowed mode I can only see 1/4 of the screen, because the render surface is bigger than the window. Can be fixed by resizing the window, but then we get mouse issues
- Mouse issues: Mouse Position out of sync when moving mouse out of window in windowed mode.
- Pixel perfect scaling: Not applied when starting in full screen

My suggestion would be to check if updating SDL will fix those issues. We might need https://github.com/ja2-stracciatella/ja2-stracciatella/pull/1500 as well, as I currently cannot build on macOS without that.

**Comment by lynxlynxlynx** [comment]
Yeah, we definitely need to bump sdl2 for the newer macs. I don't remember which release had the significant fixes, but there's no reason not to go for the latest.

**Comment by selaux** [comment]
So, after fixing those issues, should I retag so we get new release builds or do we want to include the FLTK update still?

**Comment by lynxlynxlynx** [comment]
I think it's better to first fix the fltk thing as well, otherwise we might get repeated reports of things not building. I feel it's bad form to ship in such a state.

**Comment by selaux** [comment]
There are still some remaining issues with compiling on M1 that I will have a look at.

**Comment by selaux** [comment]
So from the macOS side we should be fine now. Did anyone test the Windows builds?

**Comment by lynxlynxlynx** [comment]
No access, but I could try it through wine.

**Comment by ja2-derek** [comment]
Tested briefly (not the full Drassen test) on Windows with the 32-bit installer. It looks fine to me so far.

**Comment by ja2-derek** [comment]
@lynxlynxlynx - the "nightly builds" link in https://github.com/ja2-stracciatella/ja2-stracciatella.github.io/blob/master/_posts/_future-release-0.19.0.md is outdated.

**Comment by ja2-derek** [comment]
When upgrading from 0.18, the saved games from mods will not be accessible (after #1450). So some kind of migration has to be done - basically copying from `SavedGames-mod1-mod2` into `SavedGames`. Maybe we highlight this in the notes too?

**Comment by lynxlynxlynx** [comment]
I'll add something yeah, good catch. They can't just move them over, since they could easily overwrite other saves. Other than that, I suggest we retag and ship! :)

**Comment by selaux** [comment]
I've at least started up the win64 msvc/mingw builds on Win10 as well. They seem to work fine, didn't do the whole Drassen test though. I will retag now.

**Comment by selaux** [comment]
One think I noticed for Windows and macOS that nowadays running unsigned binaries always requires some workaround on the OSs.

- macOS needs `cmd`-key + right click and selecting "Open" in order to be able to run the binary
- Windows needs clicking on "More Options" and then clicking on "Run anyways", after double click. 

Maybe we should reflect that at some point in the [how to run guide](https://ja2-stracciatella.github.io/how-to-run/).

**Comment by lynxlynxlynx** [comment]
Not a bad idea. I tried the new appimage and all seems fine still, so I'll go ahead and make the rest of the dance.

**Comment by lynxlynxlynx** [comment]
Ok, all anouncing done, I've directly pushed the trivial version bump and this is a wrap. Great job everyone! 😃

**Comment by selaux** [comment]
Awesome :+1:. Thank you for driving this!
